scripts/data_overlap/test/test_compute_data_overlap_metrics.py

```diff
@@ -201,58 +193,91 @@ def test_create_ngram_index():
 
     assert ngram_index[5][test_5_gram] == set(
         [
-            EntryDataOverlapKey(stats_key=stats_1_key, instance_id=0, part=PART_INPUT),
-            EntryDataOverlapKey(stats_key=stats_2_key, instance_id=0, part=PART_INPUT),
-            EntryDataOverlapKey(stats_key=stats_2_key, instance_id=0, part=PART_REF),
+            EntryDataOverlapKey(stats_key=stats_1_key, instance_id="id1", index=0, part=PART_INPUT),
+            EntryDataOverlapKey(stats_key=stats_2_key, instance_id="id1", index=0, part=PART_INPUT),
+            EntryDataOverlapKey(stats_key=stats_2_key, instance_id="id1", index=0, part=PART_REF),
         ]
     )
     assert ngram_index[13][test_13_gram] == set(
         [
-            EntryDataOverlapKey(stats_key=stats_3_key, instance_id=0, part=PART_INPUT),
-            EntryDataOverlapKey(stats_key=stats_3_key, instance_id=0, part=PART_REF),
+            EntryDataOverlapKey(stats_key=stats_3_key, instance_id="id1", index=0, part=PART_INPUT),
+            EntryDataOverlapKey(stats_key=stats_3_key, instance_id="id1", index=0, part=PART_REF),
         ]
     )
 
 
-def test_compute_scenario_document_overlap():
+def test_compute_document_data_overlap():
     tokenizer = LightTokenizer()
+    stats_key_counts = dict()
     scenarios = [TEST_SCENARIO_1, TEST_SCENARIO_2]
-    all_overlap_stats: AllDataOverlapStats
     ngram_index: NgramIndex
-    all_overlap_stats = create_all_data_overlap_stats(light_scenarios=scenarios, n_values=N_VALUES)
-    ngram_index = create_ngram_index(light_scenarios=scenarios, n_values=N_VALUES, tokenizer=tokenizer)
+    ngram_index = create_ngram_index(
+        light_scenarios=scenarios, n_values=N_VALUES, tokenizer=tokenizer, stats_key_counts=stats_key_counts
+    )
 
-    compute_scenario_document_data_overlap(
+    stats_key_to_input_ids: DefaultDict[DataOverlapStatsKey, Set] = defaultdict(set)
+    stats_key_to_reference_ids: DefaultDict[DataOverlapStatsKey, Set] = defaultdict(set)
+
+    compute_document_data_overlap(
```

Comment:
This has a very weird interface (works with only `defaultdicts`, even though parameters are of type `Dict`). Either make it work with normal dicts, make this a private method (prefix with `_`), or change the method type signature (I think that DefaultDict in function signatures are weird, though).

Reply:
That's a good point, I've updated the interface to use `defaultdict`. I can make it also a private method, but we'd need to update the `compute_all_data_overlap` too or there'd be no point (as the dicts are pass through). I'm open to using `dict` instead, but would lean against it as 1) it increases code complexity 2) decreases performance (as we're bringing logic implemented in C up to python)